src/gp_activetable.c

```diff
@@ -318,17 +334,9 @@ report_active_table_helper(const RelFileNodeBackend *relFileNode)
 	item.tablespaceoid = relFileNode->node.spcNode;
 
 	LWLockAcquire(diskquota_locks.active_table_lock, LW_EXCLUSIVE);
-	entry = hash_search(active_tables_map, &item, HASH_ENTER_NULL, &found);
+	entry = ACTIVE_TABLE_ENTER(&item, &found);
```

Comment:
Why did you replace HASH_ENTER_NULL by HASH_ENTER?

Reply:
shm_hash_enter works as HASH_ENTER_NULL. with the exception that it returns NULL not when the memory has run out, but when the limit is reached.

Reply:
> it returns NULL not when the memory has run out

What is happen if memory has run out?

Reply:
we allocate the maximum number of elements when creating a table, so there should be no memory allocations during execution.

https://github.com/arenadata/gpdb/blob/215d8d8f9fec39300e4eb84977ade4fce040ebc1/src/backend/utils/hash/dynahash.c#L442-L457

Reply:
> we allocate the maximum number of elements when creating a table

Why is not used flag HASH_FIXED_SIZE in this case?

Reply:
ShmemInitHash has init_size and max_size arguments, and init_size is passed to hash_create as nelem - maximum number of elements expected, not max_size!

Reply:
in our case, `init_size == max_size`. and we should not go beyond this limit.
https://github.com/arenadata/diskquota/blob/e6899aaac6dc36d095c740d8e09c99b326a189fa/src/gp_activetable.c#L106-L107

Reply:
I think in this case we may use HASH_FIXED_SIZE flag